_integration/testsuite/install-contour-working.sh

```diff
@@ -35,42 +35,53 @@ readonly REPO=$(cd ${HERE}/../.. && pwd)
 # registry.
 readonly TAGS="main latest working"
 
-kind::cluster::exists() {
-    ${KIND} get clusters | grep -q "$1"
-}
-
-kind::cluster::load() {
-    ${KIND} load docker-image \
-        --name "${CLUSTERNAME}" \
-        "$@"
-}
+#kind::cluster::exists() {
+#    ${KIND} get clusters | grep -q "$1"
+#}
+#
+#kind::cluster::load() {
+#    ${KIND} load docker-image \
+#        --name "${CLUSTERNAME}" \
+#        "$@"
+#}
+#
+#if ! kind::cluster::exists "$CLUSTERNAME" ; then
+#    echo "cluster $CLUSTERNAME does not exist"
+#    exit 2
+#fi
+#
+## Build the current version of Contour.
+#make -C ${REPO} container IMAGE=docker.io/projectcontour/contour VERSION="v$$"
+#
+#for t in $TAGS ; do
+#    docker tag \
+#        docker.io/projectcontour/contour:"v$$" \
+#        docker.io/projectcontour/contour:$t
+#done
+#
+## Push the Contour build image into the cluster.
+#for t in $TAGS ; do
+#    kind::cluster::load docker.io/projectcontour/contour:$t
+#done
 
-if ! kind::cluster::exists "$CLUSTERNAME" ; then
-    echo "cluster $CLUSTERNAME does not exist"
-    exit 2
-fi
+# Install Contour
 
-# Build the current version of Contour.
-make -C ${REPO} container IMAGE=docker.io/projectcontour/contour VERSION="v$$"
 
-for t in $TAGS ; do
-    docker tag \
-        docker.io/projectcontour/contour:"v$$" \
-        docker.io/projectcontour/contour:$t
-done
+${KUBECTL} apply -f ${REPO}/examples/contour/00-common.yaml
 
-# Push the Contour build image into the cluster.
-for t in $TAGS ; do
-    kind::cluster::load docker.io/projectcontour/contour:$t
-done
+# Update the config file to set a GatewayAPI Gateway.
+${KUBECTL} apply -f <(sed 's/    # gateway:\n    #   name: contour\n    #   namespace: projectcontour/    gateway:\n           name: contour\n           namespace: projectcontour\n/' < ${REPO}/examples/contour/01-contour-config.yaml)
```

Comment:
we've got several spots now that are customizing the config file for int tests (fallback cert, global rate limiting, and this); I'm thinking we should just maintain a static config file YAML to use for int tests. I'll file an issue for that.

Reply:
Actually, I wouldn't expect this change here to stick, since subsequently both the install-fallback-certificate and install-ratelimit-service scripts are updating the configmap with different contents that don't include this. Though, the integration tests seem to be passing. Will look at it some more..

Reply:
Ah, I think it's due to https://github.com/projectcontour/contour/pull/3481/files#r594627493 being commented out, so the tests are actually running against `main` right now?

Reply:
Oh I didn't realized the ratelimit stuff was changing the configmap (makes sense though). If you don't mind, I may just update the ratelimit bits to sed the changes like I did for the gateway stuff to keep this moving, then can discuss like you mentioned a better/different approach.

Reply:
I'm fine with whatever works for now. I do think a static, integration test-specific file will be easiest to maintain and least convoluted in the long term but I can do that as a follow-up.

Reply:
https://github.com/projectcontour/contour/issues/3482

Reply:
So the integration tests worked? I ran them locally as well. /shrug

Reply:
Hmm, I don't understand that, the fallback & ratelimit install scripts should overwrite the config map, removing the gateway.

Reply:
Looks like the issue was that the certgen job was still set to an imagePullPolicy of Always, so when it ran it re-pulled the `main` tag from docker hub. Subsequently, when the Contour pods got created (or re-created, since the install-fallback and install-ratelimit scripts both restart the Contour deployment), they used the local `main` tag of the image, which was now the docker hub one, **not** the one built for this PR.

Reply:
OK so now the integration test is failing, as expected.